tests/bsdtestharness.c

```diff
@@ -104,9 +104,12 @@ main(int argc, char *argv[])
 	struct timeval tv_start;
 	gettimeofday(&tv_start, NULL);
 
+#ifdef HAVE_POSIX_SPAWNP
+#ifdef __APPLE__
```

Comment:
It's not clear why are using this flag - why not just always use posix_spawn() to both fork and exec, just as you do in the other change (in dispatch_io_net.c)?

Reply:
I assume you're referring to the `__APPLE__` check here - it's used to handle the Apple-only codepath where `POSIX_SPAWN_SETEXEC` can be set. The entire call isn't Apple-only. (Previously the code above defined `POSIX_SPAWN_SETEXEC` to 0 on non-Apple platforms, but I think that's confusing.)

Reply:
My question really is, why is there an Apple-only code path. Doesn't the non-Apple code path work for all platforms that have posix_spawn()? And if not, why is the code in dispatch_io_net.c different?

Reply:
The Apple-only codepath is used to start the process in a suspended state and set up monitoring before resuming it. The flags used to do this (`POSIX_SPAWN_SETEXEC` and `POSIX_SPAWN_START_SUSPENDED`) are Apple extensions. This codepath isn't necessary in dispatch_io_net because the server doesn't need to monitor the client process, and the non-Apple codepath below (`#if SIMPLE_TEST_HARNESS`) does much simpler process monitoring.